kernel/fdtable.c

```diff
@@ -73,13 +73,22 @@ int myst_fdtable_clone(myst_fdtable_t* fdtable, myst_fdtable_t** fdtable_out)
                 myst_fdops_t* fdops = entry->device;
                 void* object;
                 long r;
+                int fdflags;
 
+                /* Save the file descriptor flags */
+                fdflags = (*fdops->fd_fcntl)(fdops, entry->object, F_GETFD, 0);
+
+                /* Duplicate the object */
                 if ((r = (*fdops->fd_dup)(fdops, entry->object, &object)) != 0)
                 {
                     myst_spin_unlock(&fdtable->lock);
                     ERAISE(r);
                 }
 
+                /* Propagate the file descriptor flags to the object */
```

Comment:
With current Mystikos fd_dup() implementation for ext2fs/ramfs, the old_fd and new_fd of dup shares the same file object and its file descriptor flags.

Reply:
I modified ``fdtable.c`` so that we only perform this fdflags propagation for pipes. We will still need to revisit the treatment of CLOEXEC on files.